workbench-core/authorization/src/dynamicAuthorization/wbcGroupManagementPlugin.ts

```diff
@@ -200,6 +200,9 @@ export class WBCGroupManagementPlugin implements GroupManagementPlugin {
       if (currentStatus === 'delete_pending' && newStatus === 'active') {
         throw new ForbiddenError(`Cannot set group '${groupId}' status to active. It is pending delete.`);
       }
```

Comment:
we might want to restructure these to better account for state change. Something like:
```
throw if:
  current === undefined && new !== 'active'
OR
  current === 'active' && new !== 'delete_pending'
OR
  current === 'delete_pending' && new !== 'deleted'
```

Reply:
If we're deleting the entry from DB aren't these checks unnecessary? Once deleted `this.getGroupStatus` will throw `GroupNotFoundError` so we never get to the point to compare `currentStatus`.